app/src/main/java/org/mozilla/fenix/settings/TurnOnSyncFragment.kt

```diff
@@ -30,6 +33,11 @@ class TurnOnSyncFragment : PreferenceFragmentCompat() {
 
     override fun onResume() {
         super.onResume()
+        if (requireComponents.backgroundServices.accountManager.authenticatedAccount() != null) {
+            findNavController(this).popBackStack()
```

Comment:
Maybe return afterwards? There's no point starting the rest of initialization if we're exiting the screen. Fragment transactions aren't instant.